src/Redis.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+/*
+ * This file is part of the Cache package.
+ *
+ * Copyright (c) Daniel González
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @author Daniel González <daniel@desarrolla2.com>
+ * @author Julián Gutiérrez <juliangut@gmail.com>
+ */
+
+declare(strict_types=1);
+
+namespace Desarrolla2\Cache;
+
+use Desarrolla2\Cache\Exception\UnexpectedValueException;
+use Desarrolla2\Cache\Packer\PackerInterface;
+use Desarrolla2\Cache\Packer\SerializePacker;
+use Redis as PhpRedis;
+
+/**
+ * PHP Redis cache adapter.
+ *
+ * Errors are silently ignored but ServerExceptions are **not** caught. To PSR-16 compliant disable the `exception`
+ * option.
+ */
+class Redis extends AbstractCache
+{
+    /**
+     * @var PhpRedis
+     */
+    protected $client;
+
+    /**
+     * Redis constructor.
+     *
+     * @param PhpRedis $client
+     */
+    public function __construct(PhpRedis $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Create the default packer for this cache implementation.
+     *
+     * @return PackerInterface
+     */
+    protected static function createDefaultPacker(): PackerInterface
+    {
+        return new SerializePacker();
+    }
+
+    /**
+     * Set multiple (mset) with expire
+     *
+     * @param array    $dictionary
+     * @param int|null $ttlSeconds
+     * @return bool
+     */
+    protected function msetExpire(array $dictionary, ?int $ttlSeconds): bool
+    {
+        if (empty($dictionary)) {
+            return true;
+        }
+
+        if (!isset($ttlSeconds)) {
+            return $this->client->mset($dictionary);
+        }
+
+        $transaction = $this->client->multi();
+
+        foreach ($dictionary as $key => $value) {
+            $transaction->set($key, $value, $ttlSeconds);
+        }
+
+        $responses = $transaction->exec();
+
+        return array_reduce(
+            $responses,
+            function ($ok, $response) {
+                return $ok && $response;
+            },
+            true
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($key, $default = null)
+    {
+        $response = $this->client->get($this->keyToId($key));
+
+        return !empty($response) ? $this->unpack($response) : $default;
```

Comment:
Check as `$response !== false`. Response may be an empty string or `0` if a different serializer is used.

Reply:
According to https://www.php.net/manual/en/function.empty.php `false`, `''` and `0` are considered empty, in all those cases $default will be returned

Reply:
@juliangut `0` is a valid cached value. If `0` is cached it should be returned and not the default value.